import "resource.proto";

package SanProto;

message MonarchInfo {
  required uint64 user_id = 1;
  optional bytes name = 2;
  optional int32 level = 3;
  optional int32 exp = 4;
  optional int32 headicon_id = 5;
  optional int32 vip_level = 6;
  optional int32 union_id = 7;
  optional int32 vip_points = 8;
  optional int32 create_time = 9;
  optional bytes invite_code = 10;
  optional uint64 inviter = 11;
  optional int32 country = 12;    //1:魏 2:蜀 3:吴
  optional int32 coin = 13;      //游戏内产生的gold数
  optional int32 payment = 14;   //玩家充值的金额（元）
}

//更新主公信息
message UpdateMonarchReq {
  required MonarchInfo monarch = 1;
  optional int32 gold_cost = 2;
}
message UpdateMonarchRes{
  required int32 status = 1;
}

//查询推荐的国家势力
message QuerySuggestedCountryReq {
}
message QuerySuggestedCountryRes{
  required int32 status = 1;
  optional int32 country = 2;
  optional int32 reward_gold = 3;
}

//更新国家势力
message UpdateCountryReq {
  optional int32 old_country = 1;
  optional int32 new_country = 2;
  optional int32 item_basic_id = 3;  //后续更新势力需要消耗物品
  optional int32 item_num = 4;
  optional int32 reward_gold = 5;
}
message UpdateCountryRes{
  required int32 status = 1;   //0表示成功，1表示修改失败
  optional ResourceInfo resource = 2;
}



